src/sources/kubernetes/file_source_builder.rs

```diff
@@ -0,0 +1,222 @@
+                        + format!("{}/{}/*.log", uid, container_name).as_str())
+                    .into(),
+                );
+            }
+        }
+
+        // The following creates and adds path includes of form:
+        // LOG_DIRECTORY/namespace_*_uid/container_name/*.log
+        //
+        // To construct the paths, it's necessary to make every
+        // possible combination of uid, container_name, and namespace.
+        for uid in pod_uids {
+            for container_name in container_names {
+                for namespace in namespaces {
+                    self.file_config.include.push(
+                        (LOG_DIRECTORY.to_owned()
+                            + format!("{}_*_{}/{}/*.log", namespace, uid, container_name).as_str())
+                        .into(),
+                    );
+                }
+            }
+        }
+
+        Ok(())
+    }
+
+    /// Configures exclude in FileConfig
+    fn file_source_exclude(&mut self) {
+        // True if there is no includes
+        let no_include = self.config.include_container_names.is_empty()
+            && self.config.include_namespaces.is_empty()
+            && self.config.include_pod_uids.is_empty();
+
+        // Default excludes
+        if no_include {
+            // Since there is no user intention in including specific namespace/pod/container,
+            // exclude kuberenetes and vector logs.
+
+            // This is correct, but on best effort basis filtering out of logs from kuberentes system components.
+            // More specificly, it will work for all Kubernetes 1.14 and higher, and for some bellow that.
+            self.file_config
+                .exclude
+                .push((LOG_DIRECTORY.to_owned() + r"kube-system_*").into());
+
+            // NOTE: for now exclude images with name vector, it's a rough solution, but necessary for now
+            self.file_config
+                .exclude
+                .push((LOG_DIRECTORY.to_owned() + r"*/vector*").into());
+        }
+    }
+
+    /// Checks included uids for validity, and transforms them to numerical format.
+    fn numerical_pod_uids(&self) -> Result<Vec<String>, BuildError> {
+        let mut include_pod_uids = Vec::new();
+        for pod_uid in &self.config.include_pod_uids {
+            let mut pod_uid = pod_uid.clone();
+
+            // Check if pod_uid contains only UID valid characters
+            if !pod_uid.chars().all(|c| {
+                c.is_numeric() || ('A'..='F').contains(&c) || ('a'..='f').contains(&c) || c == '-'
+            }) {
+                error!(message = "Configuration 'include_pod_uids' contains not UID", uid = ?pod_uid);
+                return Err(BuildError::IllegalCharacterInUid {
+                    uid: pod_uid.clone(),
+                });
+            }
+
+            // Remove dashes from uid so to always have numerical format
+            // of uid.
+            pod_uid.retain(|c| c != '-');
```

Comment:
not sure I understand this one, why do we need to keep it numerical?

Reply:
To simplify handling, so we don't have to recheck, where it matters, in what format it is.